solr/core/src/java/org/apache/solr/cloud/ZkConfigSetService.java

```diff
@@ -223,6 +315,42 @@ public void downloadConfig(String configName, Path dir) throws IOException {
     }
   }
 
+  @Override
+  public List<String> listFilesInConfig(String configName) throws IOException {
+    try {
+      return zkClient.getChildren(CONFIGS_ZKNODE + "/" + configName, null, true);
+    } catch (KeeperException.NoNodeException e) {
+      return Collections.emptyList();
+    } catch (KeeperException | InterruptedException e) {
+      throw new IOException(e);
+    }
+  }
+
+  @Override
+  public List<String> listFilesInConfig(String configName, String fileName) throws IOException {
```

Comment:
You added getAllConfigsetFiles which seems to obsolete this method?  Though I like the name of this one better.